core/controllers/suggestion.py

```diff
@@ -246,8 +248,8 @@ def get(self, target_type, suggestion_type):
             suggestions = suggestion_services.get_reviewable_suggestions(
                 self.user_id, suggestion_type)
             self._render_suggestions(target_type, suggestions)
-        except Exception as e:
-            raise self.InvalidInputException(e)
+        except Exception:
+            raise self.InvalidInputException(traceback.format_exc())
```

Comment:
This will end to the client's view, right? If yes, then I don't think we should do this! Also, I'm not sure why this is under the try-except block?

Reply:
I'm not sure what you mean by why it's under the try-except block? Everything I added is within the except? But yes I think that you're right - it would be client facing - do you know how raising an exception leads to a pop up window? Is it related to AlertsService? And if so, how is AlertsService connected to raising an exception?

Reply:
> I'm not sure what you mean by why it's under the try-except block? Everything I added is within the except? 

Sorry for the misunderstanding, I meant "Why we have added the whole handler code inside try-catch block"!

for other questions:

It's written in [httpProvider's interceptor](https://github.com/oppia/oppia/blob/581be3b0e3a0043fe454af3222c1d8911a513cd5/core/templates/App.ts#L261), which works for all outgoing/incoming server calls/data.